File: backend/src/main/java/io/github/artemy/osipov/shop/service/present/PresentCommandHandler.java
package io.github.artemy.osipov.shop.service.present;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;

@Service
@RequiredArgsConstructor
public class PresentCommandHandler {

    private final PresentRepository presentRepository;

    public void on(CreatePresentCommand event) {
        Present present = new Present();
        present.setId(event.getId());
        present.setName(event.getName());
        present.setPrice(event.getPrice());
        present.setDate(LocalDateTime.now());

        event.getItems().forEach(i -> present.addItem(i.getCandyId(), i.getCount()));

        presentRepository.save(present);
    }

    public void on(RemovePresentCommand event) {
        presentRepository.deleteById(event.getId());
    }
}

File: backend/src/main/kotlin/io/github/artemy/osipov/shop/service/present/PresentCommandHandler.kt
package io.github.artemy.osipov.shop.service.present

import org.springframework.stereotype.Service
import java.time.LocalDateTime

@Service
class PresentCommandHandler(
    private val presentRepository: PresentRepository
) {
    suspend fun on(event: CreatePresentCommand) {
        val present = Present(
            id = event.id,
            name = event.name,
            price = event.price,
            createDate = LocalDateTime.now()
        )
        event.items.forEach { (candyId, count) ->
            present.addItem(candyId, count)
        }
        presentRepository.add(present)
    }

    suspend fun on(event: RemovePresentCommand) {
        presentRepository.deleteById(event.id)
    }
}